sunpy/util/scraper.py

```diff
@@ -331,3 +332,42 @@ def _smallerPattern(self, directoryPattern):
                 return None
         except Exception:
             raise
+
+    def _extractMetaURLs(self, timerange, extractor=None, translator=None):
+        """
+        Returns metadata information contained in URLs.
+
+        Parameters
+        ----------
+        timerange : `~sunpy.time.TimeRange`
+            Time interval where to find the directories for a given pattern.
+        extractor: `str`
+            Pattern to extract metadata by parsing the URL.
+        translator: `dict`
```

Comment:
What are you using the translator for out of interest?

Reply:
I'm using it for converting the things in url to what we want to show them in the table.
As for example.
In `NorHClient`, if url contains `tca` then it translates to `17*u.GHz` in the response table. It is optional param, not always required for most of the clients.

Reply:
On one hand this feels like something which should be done after the call to this method and not as a part of it. What do you think, is this sufficiently useful to warrant it adding extra complexity here?

Reply:
Agreed. We it should be a `GenericClient` method, not a part of scraper method. Since as a function, it should only return the metadata from url string and nothing else. The final dict can be formed later by using this extracted info and other fixed client attr values.